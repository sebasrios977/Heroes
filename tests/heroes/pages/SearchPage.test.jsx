import { fireEvent, render, screen } from "@testing-library/react";
import { SearchPage } from "../../../src/heroes";
import { MemoryRouter } from "react-router-dom";


const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({

    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate,
}));


describe('Pruebas en el componente <SearchPage />', () => {
    
    beforeEach(() => jest.clearAllMocks());

    test('Debe de mostrarse correctamente con valores por defecto', () => {
        
        const {container} = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        );
        
        expect(container).toMatchSnapshot();
        
    });
    
    test('Debe de mostrar a Batman y el input con el valor del queryString', () => {
        
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        );
        
        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman');
        
        const img = screen.getByRole('img');
        expect(img.src).toContain("/heroes/dc-batman.jpg");

        const divAlert = screen.getByLabelText('alert');
        expect(divAlert.style.display).toBe('none');
    });

    test('Debe de mostrar un error si no se encuentra el heroe', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman2']}>
                <SearchPage/>
            </MemoryRouter>
        );

        expect(screen.getByText('Heroe no encontrado')).toBeTruthy();
    });

    test('Debe de llamar el navigate a la pantalla nueva', () => {
        
        render(
            <MemoryRouter initialEntries={['/search?q=batman2']}>
                <SearchPage/>
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {name: 'searchText', value: 'superman'}});
        
        const form = screen.getByLabelText('form');
        fireEvent.submit(form);

        expect(mockedUseNavigate).toHaveBeenCalledWith("?q=superman");
    });
});