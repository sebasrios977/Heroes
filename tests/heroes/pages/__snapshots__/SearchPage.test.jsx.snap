// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <SearchPage /> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Página de busqueda
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Busqueda
      </h4>
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Busca un héroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Busca un heroe
      </div>
      <div
        aria-label="alert"
        class="alert alert-danger"
        style="display: none;"
      >
        Heroe no encontrado
      </div>
    </div>
  </div>
</div>
`;
